presto-main/src/main/java/com/facebook/presto/sql/TemporaryTableUtil.java

```diff
@@ -384,4 +389,59 @@ public static TableFinishNode createTemporaryTableWriteWithExchanges(
                 enableStatsCollectionForTemporaryTable ? Optional.of(aggregations.getFinalAggregation()) : Optional.empty(),
                 enableStatsCollectionForTemporaryTable ? Optional.of(statisticsResult.getDescriptor()) : Optional.empty());
     }
+
+    public static StatisticAggregations.Parts splitIntoPartialAndFinal(StatisticAggregations statisticAggregations, VariableAllocator variableAllocator, FunctionAndTypeManager functionAndTypeManager)
```

Comment:
Moved from StatisticAggregations, as it's moved to SPI which has no access to FunctionAndTypeManager